### PR TITLE
Fixes #30843 - check for seeding on template input lock

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

File: app/models/template_input.rb [hunk 1]
```diff
@@ -71,7 +71,7 @@ def prevent_delete_if_template_is_locked
   end
 
   def check_if_template_is_locked
-    if template&.locked
+    if template&.locked && !ForemanSeeder.is_seeding
```
Comment (ezr-ondrej) [inline app/models/template_input.rb:74]: What about depending on the template definition of it directly?
```suggestion
    if template&.run_temlate_changes_validation?
```

**Review by ezr-ondrej — CHANGES_REQUESTED** [review]
Good catch @jlsherrill ! :+1: 
Just a nit for IMHO bit cleaner solution.

File: app/models/template_input.rb (hunk 1 above)
Comment (jlsherrill) [reply app/models/template_input.rb:74]: aside from the typo, yes! ;)

**Review by ezr-ondrej — APPROVED** [review]
Thanks @jlsherrill ! :+1:
WWFJ

File: app/models/template_input.rb (hunk 1 above)
Comment (ezr-ondrej) [reply app/models/template_input.rb:74]: :facepalm: yeah rather using the correct method name :laughing:

Comment (ezr-ondrej) [reply app/models/template_input.rb:74]: The method is private :(

File: app/models/template_input.rb [hunk 2]
```diff
@@ -71,7 +71,7 @@ def prevent_delete_if_template_is_locked
   end
 
   def check_if_template_is_locked
-    if template&.locked
+    if template&.run_template_changes_validation?
```
Comment (ezr-ondrej) [inline app/models/template_input.rb]: I've not realized the method is private on `Template`

**Review by ezr-ondrej — CHANGES_REQUESTED** [review]

File: app/models/template_input.rb (hunk 2 above)
Comment (jlsherrill) [reply app/models/template_input.rb]: made it public

**Comment by jlsherrill** [comment]
looking into failure

**Comment by ezr-ondrej** [comment]
[test foreman] it works locally :thinking:

**Comment by jlsherrill** [comment]
worked for me too :(  but its a failure around locking and template_input.... so i think its related.

**Comment by ezr-ondrej** [comment]
> worked for me too :( but its a failure around locking and template_input.... so i think its related.

Only additional check in the method is `template.persisted?` so it would seem to me, that for some reason it's false on Jenkins? :thinking:

**Comment by jlsherrill** [comment]
@ezr-ondrej  the other one still failed, i think we'll merge this one as per discussion

**Review by ezr-ondrej — APPROVED** [review]
Thanks @jlsherrill!

**Pull request merged**